File: JS/6 kyu - Make the Deadfish swim.js
function parse(data) {
  var arr = data.split('');
  var value = 0;
  var _arr = [];
  for (var i = 0; i < arr.length; i++) {
    if (arr[i] === 'i') {
      value += 1;
    }
    if (arr[i] === 'd') {
      value -= 1;
    }
    if (arr[i] === 's') {
      value *= value;
    }
    if (arr[i] === 'o') {
      _arr.push(value);
    }
  }
  return _arr;
}

File: PYTHON/6 kyu - Make the Deadfish swim.py
def parse (data):
    val = 0
    arr = []
    for s in data:
      if s == 'i':
        val += 1
      elif s == 'd':
        val -= 1
      elif s == 's':
        val *= val
      elif s == 'o':
        arr.append(val)
      else:
        continue
    return arr;
